### PR TITLE
Add new wrappers and fix cxt issues

### PR DESCRIPTION
**Pull request by jaichopra**

### CODE REVIEW

File: qq.coffee [hunk 1]
```diff
@@ -36,7 +36,11 @@ class QqPromise
 
   done: -> @promise.done()
 
-  catch: (rejected) => @promise.catch(rejected)
+  catch: (rejected) => @then null, rejected, null
+
+  finally: (fn) => @promise.finally fn
+
+  spread: (fulfilled, rejected) => @promise.spread fulfilled, rejected
```
Comment (blambo) [inline qq.coffee]: We still need this one wrapped in the promise context

**Review by blambo — CHANGES_REQUESTED** [review]

**Comment by jaichopra** [comment]
Added new commit

**Review by blambo — APPROVED** [review]
LGTM

File: qq.coffee [hunk 2]
```diff
@@ -42,15 +42,24 @@ class QqPromise
...
+  # spread: (fulfilled, rejected) =>
+  #   return @all().then((array) =>
+  #     if fulfilled
+  #       fulfilled = @_wrap fulfilled
+  #     if rejected
+  #       rejected = @_wrap rejected
+  #     @cxt ?= process._qq_cxt
+  #     return fulfilled.apply null, array
+  #   , rejected)
+
```
Comment (blambo) [inline qq.coffee]: Let's remove the commented out code

**Review by blambo — CHANGES_REQUESTED** [review]

File: qq.coffee (hunk 2 above)
Comment (jaichopra) [reply qq.coffee]: These have been removed in the latest commit

**Comment by blambo** [comment]
LGTM

**Pull request merged**